include/eve/module/core/regular/impl/gather.hpp

```diff
@@ -25,11 +26,13 @@ gather_(EVE_SUPPORTS(cpu_), U const *ptr, wide<T, N> const& v) noexcept
   return that;
 }
 
-template<typename U, value X, integral_scalar_value T, typename N>
+template<typename U, conditional_expr C, integral_scalar_value T, typename N>
 EVE_FORCEINLINE auto
-gather_(EVE_SUPPORTS(cpu_), logical<X> const& cond, U const *ptr, wide<T, N> const& v) noexcept
+gather_(EVE_SUPPORTS(cpu_), C cx, U const *ptr, wide<T, N> const& v) noexcept
 {
-  return if_else(cond, gather(ptr, v), zero_);
+  auto src = alternative(cx, wide<U, N> {}, as<wide<U, N>> {});
+  auto m   = expand_mask(cx, as<wide<U, N>> {});
+  return if_else(m, gather(ptr, v), src);
```

Comment:
I don't think it should unconditionally load masked off elements, as these may fault.

Reply:
Good catch, will fix.